File: SmartBrowser_java_eclipse/src/com/teuskim/sbrowser/MiscUtils.java
package com.teuskim.sbrowser;

import java.io.BufferedReader;
import java.io.File;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.regex.Matcher;
import java.util.regex.Pattern;

import android.content.Context;
import android.content.res.Resources;
import android.os.Environment;

public class MiscUtils {

	public static boolean makeDirs(File dir){
		
		if( dir.exists() == false){
			return dir.mkdirs();
		}
		
		return true;
		
	}
	
	public static boolean makeDirs(String path){
		
		File f = new File(path);
		if( f.exists() == false){
			return f.mkdirs();
		}
		
		return true;
		
	}

	
	public static String getRawResourceText(Context context , int rid){
	
		Resources resource = context.getResources();
		BufferedReader reader = new BufferedReader( new InputStreamReader( resource.openRawResource(rid)));
		StringBuilder builder =  new StringBuilder();
		String line;
		try {
			
			while( ( line = reader.readLine()) != null ){
				builder.append(line);
			}
			return builder.toString();
			
		} catch (IOException e) {

		} 
		return null;
	}
	
	/**
	 * cache dir
	 * @param context
	 * @return
	 */
	public static File getImageCacheDirectory(Context context){
		
		File cacheDir;
		if(Environment.getExternalStorageState().equals(Environment.MEDIA_MOUNTED)){
			cacheDir = new File( Environment.getExternalStorageDirectory(),  "/sbrowser/cache" );
		}else{
			cacheDir = new File(context.getCacheDir(), "/sbrowser/cache");
		}
		return cacheDir;
	}
	
	/**
	 * 지정된 디렉토리 안에 내용 전부 지우기.
	 * @param f
	 */
	public static void deleteFile(File f){
		
		if(f.exists() == false)
			return;
		
		File[] list = f.listFiles();
		for( int i = 0 ; i < list.length; i++){

			if(list[i].isDirectory()){
				deleteFile(list[i]);
			}

			list[i].delete();
		}
	}
	
	
	/**
	 * email 형식 체크
	 * @param email
	 * @return
	 */
	public static boolean isValidEmail(String email){
		String regex = "^[_a-z0-9-]+(.[_a-z0-9-]+)*@(?:\\w+\\.)+\\w+$";
		Pattern p = Pattern.compile(regex);
		Matcher m = p.matcher(email);
		if( !m.matches() ) {
			return false; 
		}
		return true;
	}
	
}

File: SmartBrowser_kotlin/app/src/main/java/com/teuskim/sbrowser/MiscUtils.kt
package com.teuskim.sbrowser

import java.io.BufferedReader
import java.io.File
import java.io.IOException
import java.io.InputStreamReader
import java.util.regex.Pattern

import android.content.Context
import android.os.Environment

object MiscUtils {

    fun makeDirs(dir: File): Boolean {

        return if (dir.exists() == false) {
            dir.mkdirs()
        } else true

    }

    fun makeDirs(path: String): Boolean {

        val f = File(path)
        return if (f.exists() == false) {
            f.mkdirs()
        } else true

    }


    fun getRawResourceText(context: Context, rid: Int): String? {

        val resource = context.resources
        val reader = BufferedReader(InputStreamReader(resource.openRawResource(rid)))
        val builder = StringBuilder()
        var line: String
        try {
            line = reader.readLine()
            while (line != null) {
                builder.append(line)
                line = reader.readLine()
            }
            return builder.toString()

        } catch (e: IOException) {

        }

        return null
    }

    /**
     * cache dir
     * @param context
     * @return
     */
    fun getImageCacheDirectory(context: Context): File {

        val cacheDir: File
        if (Environment.getExternalStorageState() == Environment.MEDIA_MOUNTED) {
            cacheDir = File(Environment.getExternalStorageDirectory(), "/sbrowser/cache")
        } else {
            cacheDir = File(context.cacheDir, "/sbrowser/cache")
        }
        return cacheDir
    }

    /**
     * 지정된 디렉토리 안에 내용 전부 지우기.
     * @param f
     */
    fun deleteFile(f: File) {

        if (f.exists() == false)
            return

        val list = f.listFiles()
        for (i in list.indices) {

            if (list[i].isDirectory) {
                deleteFile(list[i])
            }

            list[i].delete()
        }
    }


    /**
     * email 형식 체크
     * @param email
     * @return
     */
    fun isValidEmail(email: String): Boolean {
        val regex = "^[_a-z0-9-]+(.[_a-z0-9-]+)*@(?:\\w+\\.)+\\w+$"
        val p = Pattern.compile(regex)
        val m = p.matcher(email)
        return if (!m.matches()) {
            false
        } else true
    }

}
